test/unittest/memory/memory_planner_validate.cpp

```diff
@@ -0,0 +1,173 @@
+                                                                     {1, 2});
+  std::vector<size_t> memory_offset;
+
+  size_t pool_size =
+    planner->planLayout(memory_size, memory_validity, memory_offset);
+
+  EXPECT_EQ(pool_size,
+            std::accumulate(memory_size.begin(), memory_size.end(), 0));
+  EXPECT_TRUE(validateNoOverlap(memory_size, memory_offset, pool_size));
+  EXPECT_TRUE(validateOverflow(memory_size, memory_offset, pool_size));
+}
+
+/**
+ * @brief Test memory planning with no overlap of requested memories
+ */
+TEST_P(MemoryPlannerValidate, none_overlap) {
+  std::mt19937 rng;
+  std::uniform_int_distribution<size_t> dist(1, MEM_BYTES);
+
+  std::vector<size_t> memory_size(MEM_QUANT);
+  for (unsigned int idx = 0; idx < memory_size.size(); idx++) {
+    memory_size[idx] = dist(rng);
+  }
+
+  std::vector<std::pair<unsigned int, unsigned int>> memory_validity(MEM_QUANT);
+  for (unsigned int idx = 0; idx < memory_validity.size(); idx++) {
+    memory_validity[idx] = {idx, idx + 1};
+  }
+
+  std::vector<size_t> memory_offset;
+  size_t pool_size =
+    planner->planLayout(memory_size, memory_validity, memory_offset);
+
+  EXPECT_TRUE(validateOverflow(memory_size, memory_offset, pool_size));
+  if (planner->getType() == nntrainer::BasicPlanner::type) {
+    EXPECT_EQ(pool_size,
+              std::accumulate(memory_size.begin(), memory_size.end(), 0));
+    EXPECT_TRUE(validateNoOverlap(memory_size, memory_offset, pool_size));
+  } else {
+    EXPECT_EQ(pool_size,
+              *std::max_element(memory_size.begin(), memory_size.end()));
+    EXPECT_TRUE(validateAllOverlap(memory_size, memory_offset));
+  }
+}
+
+/**
+ * @brief Test memory planning with partial overlap of requested memories
+ */
+TEST_P(MemoryPlannerValidate, partial_overlap) {
+  std::mt19937 rng;
+  std::uniform_int_distribution<size_t> dist(1, MEM_BYTES);
+  std::uniform_int_distribution<unsigned int> dist_interval(1, INTERVAL_SIZE);
+
+  std::vector<size_t> memory_size(MEM_QUANT);
+  for (unsigned int idx = 0; idx < memory_size.size(); idx++) {
+    memory_size[idx] = dist(rng);
+  }
+
+  std::vector<std::pair<unsigned int, unsigned int>> memory_validity(MEM_QUANT);
+  for (unsigned int idx = 0; idx < memory_validity.size(); idx++) {
+    memory_validity[idx] = {idx, idx + dist_interval(rng)};
+  }
+
+  std::vector<size_t> memory_offset;
+  size_t pool_size =
+    planner->planLayout(memory_size, memory_validity, memory_offset);
+
+  EXPECT_TRUE(validateOverflow(memory_size, memory_offset, pool_size));
+  if (planner->getType() == nntrainer::BasicPlanner::type) {
+    EXPECT_EQ(pool_size,
+              std::accumulate(memory_size.begin(), memory_size.end(), 0));
+    EXPECT_TRUE(validateNoOverlap(memory_size, memory_offset, pool_size));
+  } else {
+    EXPECT_EQ(pool_size,
+              *std::max_element(memory_size.begin(), memory_size.end()));
+    // EXPECT_TRUE(validateIntervalOverlap(memory_size, memory_offset));
```

Comment:
Is this todo?

Reply:
This will be added once other memory planners will be implemented.